src/mjolnir/graphtilebuilder.cc

```diff
@@ -191,9 +191,12 @@ GraphTileBuilder::GraphTileBuilder(const std::string& tile_dir,
 
 // Output the tile to file. Stores as binary data.
 void GraphTileBuilder::StoreTileData() {
+  std::string name{tile_dir_ + filesystem::path::preferred_separator +
+                            GraphTile::FileSuffix(header_builder_.graphid())};
```

Comment:
maybe we should modify this function to take the separator. the problem here is sometimes we use it for an actual file path and sometimes we use it for an identifier for the tile. so allowing to pass the separator here would allow both use-cases in the code

Reply:
I'll look into it, makes sense

Reply:
I changed the signature to take an optional argument and all callers where applicable. However, you're sure that's actually needed @kevinkreiser ? Would it matter if ID contain backslashes on Windows? I tried it without the conditional and at least I could build tiles and use the run_* apps. I didn't try any of this on Linux though. Would also be less changes to the code:)

Reply:
sometimes its used as a string for accessing the file system, sometimes its used as a url. at least in the latter we need to be able to always use `/` even if we are running on windows. i suppose the function has become misnamed over the years in that case!

one option would be to make change you suggested where it just always uses the preferred separator by default but then add another helper function called `GraphTile::URLSuffix` along side it which calls it with just `/`. then in `MakeSingleTileUrl` we'd call `GraphTile::URLSuffix`

Reply:
oh also id be happy to help with this!

Reply:
Ah right, somehow I forgot about the URL access, I actually did accommodate for that. Did you have a chance to look at the changes? 

`URLSuffix` would be more elegant and less implicit than `isFilePath` kwarg to `FileSuffix`. Your decision, I can integrate the changes, no worries. Thanks for the offer though:)

Reply:
@nilsnolde `isFilePath` should be fine (though id suggest using `_` rather than camal case). there is a bit of a problem here which is that we use the `facet` with a hardcoded separator of `/` to separate the numbers in the path. i think what we'd want to do is instantiate two facets above one with prefered separtor and one with `/` and then based on your `is_file_path` boolean imbue with one or the other.

Reply:
i think if you do taht you'll hardly need to change the `FileSuffix` function

Reply:
cool, thanks for the advice. the way you generate the id was a bit mysterious to me, makes more sense now though.

Reply:
should be neat enough now:)